implementations/rust/ockam/ockam_command/src/shell/mod.rs

```diff
@@ -0,0 +1,228 @@
+    no_input: bool,
+    output_format: OutputFormat,
+    mode: WriteMode,
+}
+
+/// A small wrapper around the `Write` trait, enriched with CLI
+/// attributes to facilitate output handling.
+#[derive(Clone)]
+pub struct CliStream<T: Write + Debug + Clone> {
+    writer: T,
+    no_color: bool,
+}
+
+impl<T: Write + Debug + Clone> CliStream<T> {
+    fn prepare_msg(&self, msg: &str) -> Result<String> {
+        let mut buffer = Vec::new();
+        write!(buffer, "{}", msg)?;
+        if self.no_color {
+            buffer = strip_ansi_escapes::strip(&buffer)?;
+        }
+        Ok(String::from_utf8(buffer).context("Invalid UTF-8")?)
+    }
+}
+
+pub mod mode {
+    use super::Output;
+
+    /// Write mode used when writing to the stderr stream.
+    #[derive(Clone)]
+    pub struct Logging;
+
+    /// Write mode used when writing to the stdout stream.
+    #[derive(Clone)]
+    pub struct Finished {
+        pub output: Output,
+    }
+}
+
+/// The command's output message to be displayed to the user in various formats
+#[derive(Clone)]
+pub struct Output {
+    human: Option<String>,
+    machine: Option<String>,
+    json: Option<String>,
+}
+
+impl Output {
+    fn new() -> Self {
+        Self {
+            human: None,
+            machine: None,
+            json: None,
+        }
+    }
+}
+
+/// Trait defining the main methods to write messages to the shell.
+pub trait ShellWriter: Clone {
+    fn stdout(no_color: bool) -> Self;
+    fn stderr(no_color: bool) -> Self;
+    fn is_tty(&self) -> bool;
+
+    fn write(&mut self, s: &str) -> Result<()>;
+    fn rewrite(&mut self, s: &str) -> Result<()>;
+    fn write_line(&mut self, s: &str) -> Result<()>;
+}
+
+// Core functions
+impl<W: ShellWriter> Shell<W> {
+    pub fn new(quiet: bool, no_color: bool, no_input: bool, output_format: OutputFormat) -> Self {
+        let no_color = Self::should_disable_color(no_color);
+        let no_input = Self::should_disable_user_input(no_input);
+        let stdout = W::stdout(no_color);
+        let stderr = W::stderr(no_color);
+        Self {
+            stdout,
+            stderr,
+            quiet,
+            no_input,
+            output_format,
+            mode: Logging,
+        }
+    }
+
+    pub fn can_ask_for_user_input(&self) -> bool {
+        !self.no_input && self.stderr.is_tty()
+    }
+
+    fn should_disable_color(no_color: bool) -> bool {
+        // If global argument `--no-color` is passed or the `NO_COLOR` env var is set, colors
+        // will be stripped out from output messages. Otherwise, let the terminal decide.
+        no_color || std::env::var("NO_COLOR").is_ok()
+    }
+
+    fn should_disable_user_input(no_input: bool) -> bool {
+        // If global argument `--no-input` is passed or the `NO_INPUT` env var is set we won't be able
+        // to ask the user for input.  Otherwise, let the terminal decide based on the `is_tty` value
+        no_input || std::env::var("NO_INPUT").is_ok()
+    }
+}
```

Comment:
We should look at what the conventional name for this argument is across popular command line. `--no-input` is likely not it.

Reply:
`--no-input-prompts` and `NO_INPUT_PROMPTS`

Reply:
I got that from: https://clig.dev/#interactivity

Maybe it's not widely used in general.

Reply:
I'm unable to find many concrete examples of this, outside of the link @adrianbenavides has already provided.

Most `no-[resource]` flags i've seen though have been brief, and i'd imagine given the nature of most cli flags don't lean into verbosity. 

`--no-input` would be what I'd go with.